### PR TITLE
feat(influxdb-appender):send strings values  to influx as tags instea…

### PR DESCRIPTION
**Pull request by giladw**
…d of fields

### CODE REVIEW

File: appender/influxdb/src/main/java/org/apache/karaf/decanter/appender/influxdb/InfluxDbAppender.java [hunk 1]
```diff
@@ -95,10 +96,10 @@ public void activate(Dictionary<String, Object> config) {
         String prefix = "tag.";
         for (Enumeration<String> e = config.keys(); e.hasMoreElements(); ) {
             String key = e.nextElement();
-            if (key.startsWith(prefix)) {
+            if( key.startsWith(prefix)) {
                 Object value = this.config.get(key);
                 if (value != null)
-                    tags.put(key.substring(4), value.toString());
+                    globalTags.put(key.substring(4),value.toString());
```
Comment (jbonofre) [inline appender/influxdb/src/main/java/org/apache/karaf/decanter/appender/influxdb/InfluxDbAppender.java]: I think a better name would be `tagsNames`.

File: appender/influxdb/src/main/java/org/apache/karaf/decanter/appender/influxdb/InfluxDbAppender.java [hunk 2]
```diff
@@ -120,16 +121,23 @@ public void handleEvent(Event event) {
...
                 type = (String) event.getProperty("type");
             }
-            Map<String, Object> data = new HashMap<>();
+
+            Map<String, Object> eventFields = new HashMap<>();
+            Map<String, String> eventTags = new HashMap<>();
+            eventTags.putAll(globalTags);
+
             for (String propertyName : event.getPropertyNames()) {
                 Object propertyValue = event.getProperty(propertyName);
```
Comment (jbonofre) [inline appender/influxdb/src/main/java/org/apache/karaf/decanter/appender/influxdb/InfluxDbAppender.java]: Maybe I misunderstood the config `tag.` prefix, but I think it's to "detect" property that should consider as tags.
So, basically, something like:
```
if (tagsNames.contains(propertyName) && propertyValue != null && (propertyValue instanceof String)) {
  eventTags.put(propertyName, (String) propertyValue);
}
```
Else, I don't see the purpose of `eventTags.putAll(globalTags)`.

Thoughts ?

File: appender/influxdb/src/main/java/org/apache/karaf/decanter/appender/influxdb/InfluxDbAppender.java (hunk 1 above)
Comment (giladw) [reply appender/influxdb/src/main/java/org/apache/karaf/decanter/appender/influxdb/InfluxDbAppender.java]: I wanted the difference between event tags and global tags to be explicit.
global tags will be sent with every metric.( for example application name).
event tags are specific for a single event.

File: appender/influxdb/src/main/java/org/apache/karaf/decanter/appender/influxdb/InfluxDbAppender.java (hunk 2 above)
Comment (giladw) [reply appender/influxdb/src/main/java/org/apache/karaf/decanter/appender/influxdb/InfluxDbAppender.java]: actually, at first the "tag." prefix was used for global tags.
the purpose of "eventTags.putAll(globalTags)"
is to merge the global tags and event tags.

**Comment by giladw** [comment]
hey, thank for the cr. 
I replied to the comments, and can make further changes :)

**Pull request merged**